File: Scala/src/test/scala/parrot/ParrotTest.scala
package parrot

import org.scalatest.matchers.should.Matchers
import org.scalatest.flatspec.AnyFlatSpec

class ParrotTest extends AnyFlatSpec with Matchers {

  it should "getSpeedOfEuropeanParrot" in {
    val parrot = new Parrot(ParrotType.EUROPEAN, 0, 0, false)
    parrot.getSpeed shouldEqual 12.0
  }

  it should "getSpeedOfAfricanParrot_With_One_Coconut" in {
    val parrot = new Parrot(ParrotType.AFRICAN, 1, 0, false)
    parrot.getSpeed shouldEqual 3.0
  }

  it should "getSpeedOfAfricanParrot_With_Two_Coconuts" in {
    val parrot = new Parrot(ParrotType.AFRICAN, 2, 0, false)
    parrot.getSpeed shouldEqual 0.0
  }

  it should "getSpeedOfAfricanParrot_With_No_Coconuts" in {
    val parrot = new Parrot(ParrotType.AFRICAN, 0, 0, false)
    parrot.getSpeed shouldEqual 12.0
  }

  it should "getSpeedNorwegianBlueParrot_nailed" in {
    val parrot = new Parrot(ParrotType.NORWEGIAN_BLUE, 0, 1.5, true)
    parrot.getSpeed shouldEqual 0.0;
  }

  it should "getSpeedNorwegianBlueParrot_not_nailed" in {
    val parrot = new Parrot(ParrotType.NORWEGIAN_BLUE, 0, 1.5, false)
    parrot.getSpeed shouldEqual 18.0
  }

  it should "getSpeedNorwegianBlueParrot_not_nailed_high_voltage" in {
    val parrot = new Parrot(ParrotType.NORWEGIAN_BLUE, 0, 4, false)
    parrot.getSpeed shouldEqual 24.0
  }

}

File: CSharp/Parrot.Tests/ParrotTest.cs
using Xunit;

namespace Parrot.Tests
{
    public class ParrotTest
    {
        [Fact]
        public void GetSpeedNorwegianBlueParrot_nailed()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 0, true);
            Assert.Equal(0.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedNorwegianBlueParrot_nailed_with_voltage()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 1.5, true);
            Assert.Equal(0.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedNorwegianBlueParrot_not_nailed()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 1.5, false);
            Assert.Equal(18.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedNorwegianBlueParrot_not_nailed_high_voltage()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 4, false);
            Assert.Equal(24.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedOfAfricanParrot_With_No_Coconuts()
        {
            var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 0, 0, false);
            Assert.Equal(12.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedOfAfricanParrot_With_One_Coconut()
        {
            var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 1, 0, false);
            Assert.Equal(3.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedOfAfricanParrot_With_Two_Coconuts()
        {
            var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 2, 0, false);
            Assert.Equal(0.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetSpeedOfEuropeanParrot()
        {
            var parrot = new Parrot(ParrotTypeEnum.EUROPEAN, 0, 0, false);
            Assert.Equal(12.0, parrot.GetSpeed());
        }

        [Fact]
        public void GetCryOfEuropeanParrot()
        {
            var parrot = new Parrot(ParrotTypeEnum.EUROPEAN, 0, 0, false);
            Assert.Equal("Sqoork!", parrot.GetCry());
        }

        [Fact]
        public void GetCryOfAfricanParrot()
        {
            var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 2, 0, false);
            Assert.Equal("Sqaark!", parrot.GetCry());
        }

        [Fact]
        public void GetCryNorwegianBlueParrot_high_voltage()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 4, false);
            Assert.Equal("Bzzzzzz", parrot.GetCry());
        }

        [Fact]
        public void GetCryNorwegianBlueParrot_no_voltage()
        {
            var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 0, false);
            Assert.Equal("...", parrot.GetCry());
        }
    }
}